lib/id3c/api/datastore.py

```diff
@@ -382,6 +383,63 @@ def fetch_identifier_set_uses(session: DatabaseSession) -> Any:
 
         return list(cursor)
 
+@export
+@catch_permission_denied
+def store_sample(session: DatabaseSession, sample: dict) -> Any:
+    """"
+    Validate the given *sample* and insert or update in the backing database.
+
+    Returns a list of in the same order as the input, with each object including
+    the ``sample_id`` (string), ``status`` (string) to indicate if inserted,
+    updated, or validation failed, and ``details`` to indicate reason for
+    failed validation.
+    """
```

Comment:
Much of this code seems based on id3c's `manifest` command . I imagine we might keep both around for a while. I wonder if there's a way to structure this so that the two can share the common pieces -- it might take a bit of refactoring though, because of difference in the validation messaging when we want to set a specific response elements in an API response vs logging it/skipping DETs.
If the future vision is consolidation but its too big a piece of work at this time, maybe that could be added as a TODO

Reply:
Yes, definitely potential for consolidation here. Possibly creating another general function in the `db` subpackage (as was done recently with `upsert_sample`) to handle the validation process for this and the manifest ETL. I agree that would be a good improvement, but to meet the upcoming deadline I think we should keep it as-is and tackle that later.